online-fishing-store/src/app/services/product/product.service.ts

```diff
@@ -1,27 +1,53 @@
-import { Injectable } from '@angular/core';
-import { map, Observable, of } from 'rxjs';
+import { Injectable, OnInit } from '@angular/core';
+import { BehaviorSubject, map, Observable, of, Subject } from 'rxjs';
 import { Product } from 'src/app/interfaces/product';
 import { PRODUCTS } from 'src/app/mocks/mock-products';
 import { SortService } from '../sort/sort.service';
 
 @Injectable({
   providedIn: 'root'
 })
-export class ProductService {
-public products: Product[] = PRODUCTS;
+export class ProductService{
+  public products: Product[] = PRODUCTS;
+  public products$: Subject<Product[]> = new BehaviorSubject <Product[]>(this.products);
+  public amountProductsAddedShoppingCart$: Subject<number> = new BehaviorSubject <number>(0);
```

Comment:
More right will be create productsInShippingCart subject. You will can use it in header and in shopping cart page in future. 

amountProductsAddedShoppingCart - it is redandunt method.